src/Workleap.DomainEventPropagation.Publishing/EventPropagationClient.cs

```diff
@@ -33,25 +33,30 @@ public Task PublishDomainEventAsync(string subject, IDomainEvent domainEvent, Ca
         => this.PublishDomainEventAsync(typeof(T).FullName, domainEvent, cancellationToken);
 
     public Task PublishDomainEventsAsync<T>(IEnumerable<T> domainEvents, CancellationToken cancellationToken) where T : IDomainEvent
-        => this.PublishDomainEventsAsync(typeof(T).FullName, domainEvents as IEnumerable<IDomainEvent>, cancellationToken);
+    {
+        var events = domainEvents as IEnumerable<IDomainEvent>;
+        if (events == null)
+        {
+            throw new ArgumentException("Can't cast domainEvents to IEnumerable<IDomainEvent>");
+        }
+
+        return this.PublishDomainEventsAsync(typeof(T).FullName, events, cancellationToken);
+    }
```

Comment:
I'd still leave this signature as it was before and move the argument exception to the signature below where the logic is that way this check is applied no matter which call you make